include/envoy/upstream/resource_manager.h

```diff
@@ -43,6 +44,8 @@ class Resource {
   virtual uint64_t max() PURE;
 };
 
+typedef std::shared_ptr<Resource> ResourcePtr;
```

Comment:
If we do move foward with this change, I don't think this is necessary. From reading the rest of the change, there is always some Resource available, thus we can continue to return a reference and the underlying implementation would be a unique_ptr. I would revert all of this.

Reply:
Addressed via 9e2988f.